internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -126,31 +134,59 @@ func (hr *HypershiftAgentServiceConfigReconciler) Reconcile(origCtx context.Cont
 	}
 
 	// Ensure agent-install CRDs exist on spoke cluster and clean stale ones
-	if err := hr.syncSpokeAgentInstallCRDs(ctx, spokeClient); err != nil {
+	if err = hr.syncSpokeAgentInstallCRDs(ctx, spokeClient); err != nil {
 		log.WithError(err).Error(fmt.Sprintf("Failed to sync agent-install CRDs on spoke cluster in namespace '%s'", instance.Namespace))
 		return ctrl.Result{}, err
 	}
 
 	// Initialize ASC to reconcile components according to instance context
 	asc.initHASC(hr, instance)
 
+	// Ensure relevant finalizers exist (cleanup on deletion)
+	if err = ensureFinalizers(ctx, log, asc, hypershiftAgentServiceConfigFinalizerName); err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if !instance.DeletionTimestamp.IsZero() {
+		return ctrl.Result{}, nil
+	}
+
+	// Invoke validation funcs
+	valid, err = validate(ctx, log, asc)
+	if err != nil {
+		return ctrl.Result{Requeue: true}, err
+	}
+	if !valid {
+		return ctrl.Result{}, nil
+	}
+
+	// Remove IPXE HTTP routes if not needed
+	if err = cleanHTTPRoute(ctx, log, asc); err != nil {
```

Comment:
it's either or relationship wtih the configuration of the route as far as I understand.

Reply:
Right, we check if need (exposeIPXEHTTPRoute) in the cleanHTTPRoute func.